compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -82,7 +83,7 @@ pub fn obligations<'tcx>(
 pub fn trait_obligations<'tcx>(
     infcx: &InferCtxt<'tcx>,
     param_env: ty::ParamEnv<'tcx>,
-    body_id: hir::HirId,
+    body_id: def_id::LocalDefId,
```

Comment:
```suggestion
    body_id: LocalDefId,
```
qualified `DefId`/`LocalDefId`/`CRATE_DEF_ID` may deserve an internal lint...

Reply:
I did not get your suggestion sorry, can you clarify?

Reply:
I mean: there is an internal lint when someone write 'ty::TyCtxt' instead of just TyCtxt. We could use a similar one around DefId and co. 
(it's just a passing thought, not for this PR)